daemon/logic/engine.go

```diff
@@ -274,64 +281,74 @@ func (e *Engine) RetrieveCredentials(ctx context.Context,
 			keypairs[*orgID] = kp
 		}
 
-		krm, mekshare, err := graph.FindMember(e.session.AuthID())
-		if err != nil {
-			log.Printf("Error finding keyring membership: %s", err)
-			return nil, err
-		}
-
-		encryptingKey, ok := encryptingKeys[*krm.EncryptingKeyID]
+		encryptingKey, ok := encryptingKeys[encryptingKeyID]
 		if !ok {
 			encryptingKey, err = findEncryptingKey(ctx, e.client, orgID,
-				krm.EncryptingKeyID)
+				&encryptingKeyID)
 			if err != nil {
 				log.Printf("Error finding encrypting key for user: %s", err)
 				return nil, err
 			}
-			encryptingKeys[*krm.EncryptingKeyID] = encryptingKey
+			encryptingKeys[encryptingKeyID] = encryptingKey
 		}
 
-		err = e.crypto.WithUnboxer(ctx, *mekshare.Key.Value, *mekshare.Key.Nonce, &kp.Encryption, *encryptingKey.Key.Value, func(u crypto.Unboxer) error {
-			for _, cred := range graph.GetCredentials() {
-				pt, err := u.Unbox(ctx, *cred.Credential().Value, *cred.Nonce(), *cred.Credential().Nonce)
+		err = e.crypto.WithUnsealer(ctx, &kp.Encryption, *encryptingKey.Key.Value, func(unsealer crypto.Unsealer) error {
```

Comment:
One downside is that we actually have the unsealed private key sitting around for longer inside memory. Something to think about.